api/src/test/java/org/openmrs/EncounterTest.java

```diff
@@ -1397,4 +1420,40 @@ public void hasDiagnosis_shouldReturnFalseIfEncounterDoesNotHaveDiagnosis(){
 		assertFalse(encounter.hasDiagnosis(diagnosis));
 	}
 
+	/**
+	 * @see Encounter#getConditions()
+	 */
+	@Test
+	public void getConditions_shouldReturnAllConditionsWhenFlagIsTrue() {
+
+		assertEquals(2, encounter.getConditions(true).size());
+		assertTrue(encounter.getConditions(true).contains(activeCondition));
+		assertTrue(encounter.getConditions(true).contains(voidedCondition));
+	}
+
+	/**
+	 * @see Encounter#getConditions()
+	 */
+	@Test
+	public void getConditions_shouldReturnNonVoidedConditionsByDefault() {
+
+		assertEquals(1, encounter.getConditions().size());
+		assertTrue(encounter.getConditions().contains(activeCondition));
+		assertFalse(encounter.getConditions().contains(voidedCondition));
+	}
+
+	/**
+	 * @see Encounter#removeCondition(Condition)
+	 */
+	@Test
+	public void removeCondition_shouldRemoveConditions() {
+
+		// The condition should be voided
+		encounter.removeCondition(activeCondition);
+
+		Set<Condition> a = encounter.getConditions();
```

Comment:
Rename `a` to `conditions` or `allConditions`.
Rename `b` to `voidedConditions`.